src/common/types/store-data/card-selection-store-data.ts

```diff
@@ -1,17 +1,11 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
-export interface CardSelectionData {
-    resultInstanceUid: string;
-    isSelected: boolean;
-}
-
 export interface RuleExpandCollapseData {
-    ruleId: string;
     isExpanded: boolean;
-    cards: CardSelectionData[];
+    cards: { [resultInstanceUid: string]: boolean };
```

Comment:
You can still declare a type with this and use that here.

```export type CardSelectionData = { [resultInstanceUid: string]: boolean };```

and then reference accordingly. Same goes for below.

Reply:
Done.